### DISCUSSION TITLE
Co-adviser and Glossaries

### DISCUSSION THREAD

**Discussion by Martafct** [joaomlourenco/novathesis#401] [Q&A] [answered]
Hello. First I just wanted to thank you for all your work and help. I just have some more questions. 

After applying the Bsc template, I tried to add the co-adviser name and it does not appear, but the adviser does. When I was using the Msc template or any other, both of the names would appear. I compared both files, and I don't see a difference. Here I show a screenshot of the 3_cover.tex. 

Finally, I'm having trouble with the glossaries, acronyms and symbols. For the last few months I've been trying to get it right, to not bother you, but I just can't do it. For glossaries, for example, I added an entry, so the file template.glo is not empty when I opened it. And I added the screenshot of just an example of \gls that I did. But then, I tried to add automake in packages and gives error. I searched on the internet and I tried everything and also gives error. I am not sure if I have to create the template.gls, etc, or I'm missing something. Bellow there is also screenshots of the packages.tex with and without automake.

Thank you. 

<img width="1292" alt="cover" src="https://user-images.githubusercontent.com/115926172/209857970-9b5744f4-8f51-4a4b-8955-0fe0844a4152.png">
<img width="1287" alt="automake" src="https://user-images.githubusercontent.com/115926172/209857976-788bee0b-560e-42a1-8b00-2a3f2c5108b8.png">
<img width="1287" alt="semautomake" src="https://user-images.githubusercontent.com/115926172/209857980-b9bc4e26-502e-43df-b0da-422cfc9f5f8a.png">
<img width="1287" alt="glossaries" src="https://user-images.githubusercontent.com/115926172/209857991-88fd9e73-0075-4b26-82ab-ebb507651605.png">
<img width="1287" alt="glsintroducao" src="https://user-images.githubusercontent.com/115926172/209857993-798a3b8d-93f6-4afb-be6d-bbae19bf6688.png">

**Comment by joaomlourenco** [comment]
Hi @Martafct 
Unfortunately, for a while there was a version available in the repository with a bug, and it would only print the adviser (and not the co-advisers).  Probably you are using that version.  Please update to the latest version in the repository.  It should work all right (I just tested and it did work all right for me).

Concerning the glossaries, I just used TeXstudio to compile the "standard" project and it properly produces all the glossaries.  Please try it in a "vanila" version downloaded from the repository and check if it works.  If it does, that means you have a problem in your definitions (check the file "template.log" and search for errors).  If it does not, please confirm that in your "Preferences->Build->Default Compiler" you have "Latexmk".

**Reply by Martafct** [reply]
Hello João. Well, I updated to latest version, but now, the document is not opening. I don't know why.... 
And I tried with the latest version of Texstudio also. Bellow there is a screenshot of what appears. 
<img width="1287" alt="Captura de ecrã 2023-01-05, às 17 42 41" src="https://user-images.githubusercontent.com/115926172/210847060-758616ac-363d-4e6f-b4cd-276c5a145c64.png">
<img width="1287" alt="Captura de ecrã 2023-01-05, às 17 42 58" src="https://user-images.githubusercontent.com/115926172/210847072-73e16507-e473-425f-8dc2-0f53d340d0a2.png">


Can you see what is the problem? Thank you!

**Comment by joaomlourenco** [comment, marked as answer]
Which version of LaTeX are you using?
Open the "TeX Live Utility", then go to menu "Configure" and then "Change Default TeX Live Version..." and tell me which version(s) you have there.

**Reply by joaomlourenco** [reply]
Just to be sure I following the situation…
You downloaded the most recent version from github, you *did not change* anything at all, tried to compile the document and got some errors!  Is that it?

**Reply by Martafct** [reply]
Exactly. I downloaded the latest version to try to see if there was the problem with co-advisors. But I did not change anything. I just download it and compiled it. And I got the errors that I showed in the screenshot above.

**Reply by joaomlourenco** [reply]
That's so weired. It works for me with the same LaTeX…

Can you please "copy&paste" here the contents of your line 23 in file "3_cover.tex"? Thanks.

**Reply by Martafct** [reply]
Line 23? It just appears this: 

\nttitle(main,pt){Um Título de Tese Longo\\e com uma Mudança de Linha Forçada}%

I'm not sure if this what you wanted.

**Reply by joaomlourenco** [reply]
Can't understand the problem and can't reproduce it locally.  
I'm assuming you have your LaTeX installation up to date (using Tex Live Utility). Am I correct?
Can you please prepare a ZIP with your document folder and send it to me (FB messenger, email, …)?  And then I'll try with the same source… if it works, the problem is in your LaTeX installation.  If it does not, then I'll figure out what's the problem.

**Reply by Martafct** [reply]
Hello again. I'm sorry to bother you again. I guess the problem is not resolved. I was writing the title of my project, and it got erros again. But when I wrote it without "acentos gráficos", it was okay. So right now I can't write the title with them. Do you know any way to resolve this? Thank you. 
<img width="1400" alt="Captura de ecrã 2023-01-10, às 10 44 01" src="https://user-images.githubusercontent.com/115926172/211531391-de4470ea-d28c-4edf-94cb-58e68817d092.png">
<img width="1400" alt="Captura de ecrã 2023-01-10, às 10 44 18" src="https://user-images.githubusercontent.com/115926172/211531407-47051527-77c4-4363-adc5-baee1ba860f9.png">

**Reply by Martafct** [reply]
Besides that, I'm trying to add Acronyms and etc. And I tried to only change the ones you added on the template. But the minute I do that, I get erros. And I change the words in the chapter and the .tex. I don't understand why I'm having so many problems.

**Reply by joaomlourenco** [reply]
@Martafct 
I was able to reproduce the problem in a friend's computer.  For some mysterious reason, it worked all right with mine, but not in his and the symptoms were the same you have.

I believe the easiest solution is to go to the repository in github, download the file `novathesis.cls` e replace yours.  I believe it should work.  If it does not, then you need download everything, and then replace the folders `Config` and `Chapters` with yours.

Please let me know if it worked.

**Comment by Martafct** [comment]
It just appears Texlive 2022. 
<img width="652" alt="Captura de ecrã 2023-01-05, às 20 26 08" src="https://user-images.githubusercontent.com/115926172/210873775-43bb362d-2b5d-48f9-913a-b460022c920d.png">

**Reply by joaomlourenco** [reply]
@Martafct
I was able to reproduce the problem in a friend's computer. For some mysterious reason, it worked all right with mine, but not in his and the symptoms were the same you have.

I believe the easiest solution is to go to the repository in github, download the file novathesis.cls e replace yours. I believe it should work. If it does not, then you need download everything, and then replace the folders Config and Chapters with yours.

Please let me know if it worked.

**Reply by Martafct** [reply]
Yeah that's strange. Yes, everything is working out okay now. Thank you :).
